### PR TITLE
Added subroutine to run an NLTK Stemmer

### PR DESCRIPTION
**Pull request by nitkul**
So I saw an issue about TextBlob using a lemmatizer but not a stemmer.
Wrote a function for the later.
@sloria

### CODE REVIEW

File: textblob/blob.py [hunk 1]
```diff
@@ -148,6 +148,16 @@ def lemmatize(self, pos=None):
         lemmatizer = nltk.stem.WordNetLemmatizer()
         return lemmatizer.lemmatize(self.string, pos)
 
+    #added 'stemmer' on lines of lemmatizer
+    #based on nltk
+    def stem(self, isporter=True):
+        #param isporter: True when using Porter stemmer, else Snowball
```
Comment (sloria) [inline textblob/blob.py]: Please document this method with a docstring, using the Sphinx syntax (see other methods for examples).

File: textblob/blob.py [hunk 2]
```diff
@@ -148,6 +148,16 @@ def lemmatize(self, pos=None):
         lemmatizer = nltk.stem.WordNetLemmatizer()
         return lemmatizer.lemmatize(self.string, pos)
 
+    #added 'stemmer' on lines of lemmatizer
```
Comment (sloria) [inline textblob/blob.py]: These comments are unnecessary; remove them.

File: textblob/blob.py [hunk 3]
```diff
@@ -148,6 +148,16 @@ def lemmatize(self, pos=None):
...
+    #based on nltk
+    def stem(self, isporter=True):
```
Comment (sloria) [inline textblob/blob.py]: Are there other stemmers available in NLTK? If so, I'm not sure a boolean param will suffice here. Perhaps the user could pass an instance of a stemmer, e.g.

```python
blob.stem(stemmer=PorterStemmer())
```

File: tests/test_blob.py [hunk 4]
```diff
@@ -914,6 +918,14 @@ def test_lemma(self):
         w = tb.Word("went", "VBD");
         assert_equal(w.lemma, "go")
 
+    def test_stem(self): #only PorterStemmer tested
+        w = tb.Word("cars")
+        assert_equal(w.stem(), "car")
+        w = tb.Word("wolves")
+        assert_equal(w.stem(), "wolv")
+        w = tb.Word("went")
+        assert_equal(w.stem("v"), "went")
```
Comment (sloria) [inline tests/test_blob.py]: This assertion is failing because of the invalid `"v"` argument.

**Comment by nitkul** [comment]
thanks man!
i've pushed changes. i also altered SentenceTokenizer() - i wasn't
particularly aiming for that, just hoping to solve an issue - but those
changes are harmless

Regards,


*Nitish Kulshrestha | +91 915 866 4202*
Core Member
Alumni Affairs Division





*BITS Pilani K. K. Birla Goa CampusNH17-B, ZuarinagarGoa- 403726India*

**Comment by sloria** [comment]
The tokenization changes are unrelated to this PR; please remove them.

**Comment by sloria** [comment]
The added test is failing; can you please look into it?

**Comment by nitkul** [comment]
@sloria I have fixed errors in the stemmer, rest seem to be errors in TextBlob itself. Please look into it.

**Comment by sloria** [comment]
Yes, the errors are unrelated to this PR. Thanks for making those changes. This looks good to go.

**Pull request merged**